cli/tests/services.bats

```diff
@@ -1091,14 +1091,40 @@ EOF
   done
 }
 
+@test "watchdog: emits heartbeat log to prevent garbage collection while running" {
+  target_pid="$$"
+  registry_file="$PWD/registry.json"
+  dummy_registry path/to/env abcde123 > "$registry_file"
+  _FLOX_WATCHDOG_LOG_LEVEL=debug "$WATCHDOG_BIN" \
+    --log-dir "$PWD" \
+    --pid "$target_pid" \
+    --registry "$registry_file" \
+    --hash abcde123 \
+    --socket does_not_exist &
+
+  watchdog_pid="$!"
+  log_file="$PWD/watchdog.${target_pid}.log"
+
+  # Wait for initial log entry. Other entries will be printed later but we don't
+  # want to wait that long.
+  timeout 1s bash -c "
+    while ! grep -qs 'still watching, woof woof' \"$log_file\"; do
+      sleep 0.1
+    done
+  "
+
+  # Signal to exit.
+  run kill "$watchdog_pid"
+  assert_success
```

Comment:
I wonder, does bats run the `kill` even if the timeout failed? or does it abort and directly run the teardown?
if its the latter that might leave a watchdog child open (which would have a fd3 as well i assume🤔)

Reply:
maybe at least close fd3 for the watchdog process then, to avoid the chance of this test failing to block the entire testsuite

Reply:
Done